src/metabase/models/setting.clj

```diff
@@ -479,10 +522,7 @@
          :as setting}                     (resolve-setting setting-definition-or-name)
         obfuscated?                       (and sensitive? (obfuscated-value? new-value))
         setting-name                      (setting-name setting)]
-    ;; make sure we're not trying to set the value of a Database-local-only Setting or something like that
-    (when-not (allows-site-wide-values? setting)
-      (throw (ex-info (tru "Site-wide values are not allowed for Setting {0}" (:name setting))
-                      {:setting (:name setting)})))
```

Comment:
Moved this below into the `do` block, so that we _can_ set the value of a User-local setting